docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+        import org.apache.kafka.streams.kstream.GlobalKTable;
+
+	        KStreamBuilder builder = new KStreamBuilder();
+
+	        GlobalKTable&lt;String, Long&gt; wordCounts = builder.globalTable(
+		        Serdes.String(), /* key serde */
+		        Serdes.Long(),   /* value serde */
+		        "word-counts-input-topic", /* input topic */
+		        "word-counts-global-store" /* table/store name */);
+	    </pre>
+	    
+	    When to provide serdes explicitly:
+	    <ul>
+	    <li>If you do not specify serdes explicitly, the default serdes from the configuration are used.</li>
+	    <li>You must specificy serdes explicitly if the key and/or value types of the records in the Kafka input topic do not
+	      match the configured default serdes.</li>
+	    
+	    Several variants of globalTable exist to e.g. specify explicit serdes.
+
+	</td>
+  </tbody>
+  </table>
 
-    KStream&lt;String, GenericRecord&gt; source1 = builder.stream("topic1", "topic2");
-    KTable&lt;String, GenericRecord&gt; source2 = builder.table("topic3", "stateStoreName");
-    GlobalKTable&lt;String, GenericRecord&gt; source2 = builder.globalTable("topic4", "globalStoreName");
-    </pre>
+    <h4><a id="streams_dsl_transform" href="#streams_dsl_transform">Transform a stream</a></h4>
 
-    <h4><a id="streams_dsl_windowing" href="#streams_dsl_windowing">Windowing a stream</a></h4>
+    <code>KStream</code> and <code>KTable</code> support a variety of transformation operations. Each of these operations can be translated into one or more
+    connected processors into the underlying processor topology. Since <code>KStream</code> and <code>KTable</code> are strongly typed, all these transformation
+    operations are defined as generics functions where users could specify the input and output data types.
+
+    Some <code>KStream</code> transformations may generate one or more <code>KStream</code> objects (e.g., filter and map on <code>KStream</code> generate another <code>KStream</code>,
+    while branch on <code>KStream</code> can generate multiple <code>KStream</code>) while some others may generate a <code>KTable</code> object (e.g., aggregation) interpreted
+    as the changelog stream to the resulted relation. This allows Kafka Streams to continuously update the computed value upon arrivals
+    of late records after it has already been produced to the downstream transformation operators. As for <code>KTable</code>, all its transformation
+    operations can only generate another <code>KTable</code> (though the Kafka Streams DSL does provide a special function to convert a <code>KTable</code>
+    representation into a <code>KStream</code>, which we will describe later). Nevertheless, all these transformation methods can be chained
+    together to compose a complex processor topology.
+
+    We describe these transformation operations in the following subsections, categorizing them into two categories:
```

Comment:
This paragraph needs to be wrapped into a `<p>` or sth -- it displayed as a single paragraph together with the one before.